src/schema.rs

```diff
@@ -845,8 +918,67 @@ mod impls {
                     let func_name = receive_name.entrypoint_name();
                     Ok(json!({"contract": contract_name, "func": func_name}))
                 }
+                Type::ULeb128(constraint) => {
+                    let int = deserial_biguint(source, *constraint)?;
+                    Ok(Value::String(int.to_string()))
+                }
+                Type::ILeb128(constraint) => {
+                    let int = deserial_bigint(source, *constraint)?;
+                    Ok(Value::String(int.to_string()))
+                }
+                Type::ByteList(size_len) => {
+                    let len = deserial_length(source, *size_len)?;
+                    let mut string = String::new();
+                    for _ in 0..len {
+                        let byte = source.read_u8()?;
+                        string.push_str(&format!("{:02X?}", byte));
+                    }
+                    Ok(Value::String(string))
+                }
+                Type::ByteArray(len) => {
+                    let len = (*len).try_into()?;
```

Comment:
What is this for? What is the type of `len` you are assigning to?

Reply:
Not sure what you are asking.
`len` is used by the for loop, but now also to preallocate the string

Reply:
Yes, but I did not undersand why you did the type conversion 
```
let len = (*len).try_into()?;
```
instead of just using `*len` in the loop bound.

It was unclear to me what the type of the new len was either.

I see now you changed it to `usize::try_from` which is clear. It also makes sense now since the new len is used in the with_capacity function which needs a usize.